traffic_ops/app/lib/API/Configs/ApacheTrafficServer.pm

```diff
@@ -1451,24 +1449,51 @@ sub logging_dot_config {
 			$text .= "	Format = '" . $format . " '\n";
 			$text .= "}\n";
 		}
+	}
+
+	for ( my $i = 0; $i < $max_log_objects; $i = $i + 1) {
+		my $log_filter_field = "LogFilter";
+		if ( $i > 0 ) {
+			$log_filter_field = $log_filter_field . "$i";
+		}
+		my $log_filter_name = $data->{$log_filter_field . ".Name"} || "";
+		if ( length($log_filter_name) > 0) {
+			my $filter = $data->{$log_filter_field . ".Filter"};
+			my $log_filter_type = $data->{$log_filter_field . ".Type"} || "accept";
+			$text .= $log_filter_name . " = filter." . $log_filter_type . "('" . $filter . "')\n";
```

Comment:
I like this much better. It's got one tiny issue left, I think. If `$filter` contains any single-quotes or backslashes, they will need to be escaped.

Also, I went back and checked. The only reason we're escaping double-quotes in the `$format` is that we used to put `$format` into an XML string that used quotes. We now harmlessly but pointlessly escape double quotes in the `$format`, but fail to usefully escape the single-quotes we now use.

Reply:
Added some escaping, although I'm really tempted to say "Don't do that" when it comes to using single quotes or backslashes in log filters. . . .